File: CPP/Queue.cpp
#include<iostream>
using namespace std;

struct node 
{
    int data;
    struct node *next;
};

class QueueX
{
private:
    struct node *first;
    int iCount;

public:
    QueueX();        
    void enqueue(int no);      
    void dequeue();               
    void Display(); 
    int Count();             
};

QueueX::QueueX()
{
    first = NULL;
    iCount = 0;
}

void QueueX::enqueue(int no)
{
    struct node *newn = new node();
    newn->data = no;
    newn->next = NULL;

    if (first == NULL)
    {
        first = newn;
    }
    else
    {
        struct node *temp = first;
        while (temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }

    iCount++;
}

void QueueX::dequeue()
{
    if (first == NULL)
    {
        cout << "Queue is empty. Cannot dequeue.\n";
        return;
    }

    struct node *temp = first;

    first = first->next;
    delete temp;
    iCount--;
}

void QueueX::Display()
{
    struct node *temp = first;

    while (temp != NULL)
    {
        cout << "| " << temp->data << " | -> ";
        temp = temp->next;
    }
}

int QueueX::Count()
{
    return iCount;
}

int main()
{
    QueueX *qobj = new QueueX();
    int iRet = 0;

    qobj->enqueue(51);
    qobj->enqueue(21);
    qobj->enqueue(11);
    qobj->enqueue(101);
    qobj->enqueue(71);

    qobj->Display();

    iRet = qobj->Count();
    cout << "Number of elements in the queue: " << iRet << "\n";

    qobj->dequeue();
    qobj->dequeue();
    qobj->dequeue();

    qobj->Display();

    iRet = qobj->Count();
    cout << "Number of elements in the queue: " << iRet << "\n";

    delete qobj;

    return 0;
}

File: Java/Queue.java
class node 
{
    public int data;
    public node next;
};

class QueueX
{
    private node first;
    private int iCount;

    public QueueX()
    {
       this.first = null;
       this.iCount = 0;
    }

    public void enqueue(int no)   //// InsertLast 
    {
       node newn = null;
       
       newn = new node();

       newn.data = no;
       newn.next = null;

       if(first == null)
       {
          first = newn;
       }
       else
       {
           node temp = first;
          
           while(temp.next != null)
           {
             temp = temp.next;
           }

           temp.next = newn; 
       }
       iCount++;
    }

    public int  dequeue()    //Deletefirst
    {
        if(first == null)
        {
            System.out.println("Unable to dequeue in queue :");
            return -1;
        }
        else
        {
             node temp = first;

             first = first.next;
             iCount--;
             return temp.data;
        }
    }

    public void Display()
    {
        node temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" | -");
            temp = temp.next;
        }
    }

    public int Count()
    {
        return iCount;
    }

    
}

public class Queue 
{
    public static void main(String[] args) 
    {
        QueueX sobj = new QueueX();
        int iRet = 0;

        sobj.enqueue(10);
        sobj.enqueue(20);
        sobj.enqueue(30);
        sobj.enqueue(40);
        sobj.enqueue(50);
        sobj.enqueue(60);

        sobj.Display();
        iRet = sobj.Count();

        System.out.println("Number of the element queue are :" + iRet);

        iRet = sobj.dequeue();
        System.out.println("Remove the element form queue :"+ iRet);

        iRet = sobj.dequeue();
        System.out.println("Remove the element form queue :"+ iRet);

        iRet = sobj.dequeue();
        System.out.println("Remove the element form queue :"+ iRet);

        sobj.Display();
        iRet = sobj.Count();

        System.out.println("Number of the element queue are :" + iRet);
    }
}
